Vector.c

```diff
@@ -18,9 +18,10 @@ Vector* Vector_new(const ObjectClass* type, bool owner, int size) {
 
    if (size == DEFAULT_SIZE)
       size = 10;
+   assert(size > 0);
    this = xMalloc(sizeof(Vector));
    this->growthRate = size;
-   this->array = (Object**) xCalloc(size, sizeof(Object*));
+   this->array = xMalloc(size * sizeof(Object*));
```

Comment:
Please overflow-check this multiplication.

Reply:
dropped this change; it seems not worth it after all